tests/storage/clone.go

```diff
@@ -447,6 +450,114 @@ var _ = SIGDescribe("[Serial]VirtualMachineClone Tests", func() {
+						}
+
+						if vm.Spec.Preference.RevisionName == "" {
+							return fmt.Errorf("preference revision name is expected to not be empty")
+						}
+						return nil
+					}
+				}
+
+				BeforeEach(func() {
+					snapshotStorageClass, err := getSnapshotStorageClass(virtClient)
+					Expect(err).ToNot(HaveOccurred())
+
+					if snapshotStorageClass == "" {
+						Skip("Skipping test, no VolumeSnapshot support")
+					}
+
+					instancetype = &instancetypev1alpha2.VirtualMachineInstancetype{
+						ObjectMeta: metav1.ObjectMeta{
+							GenerateName: "vm-instancetype-",
+							Namespace:    util.NamespaceTestDefault,
+						},
+						Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+							CPU: instancetypev1alpha2.CPUInstancetype{
+								Guest: 1,
+							},
+							Memory: instancetypev1alpha2.MemoryInstancetype{
+								Guest: resource.MustParse("128Mi"),
+							},
+						},
+					}
+					instancetype, err := virtClient.VirtualMachineInstancetype(util.NamespaceTestDefault).Create(context.Background(), instancetype, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					preference = &instancetypev1alpha2.VirtualMachinePreference{
+						ObjectMeta: metav1.ObjectMeta{
+							GenerateName: "vm-preference-",
+							Namespace:    util.NamespaceTestDefault,
+						},
+						Spec: instancetypev1alpha2.VirtualMachinePreferenceSpec{
+							CPU: &instancetypev1alpha2.CPUPreferences{
+								PreferredCPUTopology: instancetypev1alpha2.PreferSockets,
+							},
+						},
+					}
+					preference, err := virtClient.VirtualMachinePreference(util.NamespaceTestDefault).Create(context.Background(), preference, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					sourceVM = tests.NewRandomVMWithDataVolumeWithRegistryImport(
+						cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine),
+						util.NamespaceTestDefault,
+						snapshotStorageClass,
+						k8sv1.ReadWriteOnce,
+					)
+
+					sourceVM.Spec.Template.Spec.Domain.Resources = virtv1.ResourceRequirements{}
+					sourceVM.Spec.Instancetype = &virtv1.InstancetypeMatcher{
+						Name: instancetype.Name,
+						Kind: "VirtualMachineInstanceType",
+					}
+					sourceVM.Spec.Preference = &virtv1.PreferenceMatcher{
+						Name: preference.Name,
+						Kind: "VirtualMachinePreference",
+					}
+
+					sourceVM, err = virtClient.VirtualMachine(sourceVM.Namespace).Create(sourceVM)
+					Expect(err).ToNot(HaveOccurred())
+
+					for _, dvt := range sourceVM.Spec.DataVolumeTemplates {
+						libstorage.EventuallyDVWith(sourceVM.Namespace, dvt.Name, 180, HaveSucceeded())
+					}
+
+				})
+
+				It("should create new ControllerRevisions for cloned VM", Label("instancetype", "clone"), func() {
+					By("Waiting until the source VM has instancetype and preference RevisionNames")
+					Eventually(checkForVMInstancetypeRevisionNames(util.NamespaceTestDefault, sourceVM.Name), 300*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
```

Comment:
nit: you can introduce a `waitForVMInstanceTypeRevisionNames(vmName string)` that would wrap this up nicely.

Reply:
Yup fair point, updated.